openmetadata-service/src/main/java/org/openmetadata/service/secrets/SecretsManager.java

```diff
@@ -202,82 +209,102 @@ public OpenMetadataConnection decryptOpenMetadataConnection(OpenMetadataConnecti
         decryptPasswordFields(openMetadataConnectionConverted);
       } catch (Exception e) {
         throw new CustomExceptionMessage(
-            Response.Status.BAD_REQUEST, "Failed to encrypt OpenMetadataConnection instance.");
+            Response.Status.BAD_REQUEST, "Failed to decrypt OpenMetadataConnection instance.");
       }
       return openMetadataConnectionConverted;
     }
     return null;
   }
 
   private Object encryptPasswordFields(Object toEncryptObject, String secretId, boolean store) {
-    if (!DO_NOT_ENCRYPT_CLASSES.contains(toEncryptObject.getClass())) {
+    try {
+      if (!DO_NOT_ENCRYPT_CLASSES.contains(toEncryptObject.getClass())) {
+        // for each get method
+        Arrays.stream(toEncryptObject.getClass().getMethods())
+            .filter(ReflectionUtil::isGetMethodOfObject)
+            .forEach(
+                method -> {
+                  Object obj = ReflectionUtil.getObjectFromMethod(method, toEncryptObject);
+                  String fieldName = method.getName().replaceFirst("get", "");
+                  // if the object matches the package of openmetadata
+                  if (obj != null && obj.getClass().getPackageName().startsWith("org.openmetadata")) {
+                    // encryptPasswordFields
+                    encryptPasswordFields(
+                        obj, buildSecretId(false, secretId, fieldName.toLowerCase(Locale.ROOT)), store);
+                    // check if it has annotation
+                  } else if (obj != null && method.getAnnotation(PasswordField.class) != null) {
+                    // store value if proceed
+                    String newFieldValue =
+                        storeValue(fieldName, fernet.decryptIfApplies((String) obj), secretId, store);
+                    // get setMethod
+                    Method toSet = ReflectionUtil.getToSetMethod(toEncryptObject, obj, fieldName);
+                    // set new value
+                    ReflectionUtil.setValueInMethod(
+                        toEncryptObject,
+                        Fernet.isTokenized(newFieldValue)
+                            ? newFieldValue
+                            : store ? fernet.encrypt(newFieldValue) : newFieldValue,
```

Comment:
we need to remove fernet.encrypt from here and make it into DBSecretesManager and except that logic to live inside of these secret managers